src/ambulance_game/game.py

```diff
@@ -353,7 +355,149 @@ def get_routing_matrix(
+        mu_1=mu_1,
+        mu_2=mu_2,
+        num_of_servers_1=num_of_servers_1,
+        num_of_servers_2=num_of_servers_2,
+        system_capacity_1=system_capacity_1,
+        system_capacity_2=system_capacity_2,
+        buffer_capacity_1=buffer_capacity_1,
+        buffer_capacity_2=buffer_capacity_2,
+        threshold_1=threshold_1,
+        threshold_2=threshold_2,
+        alpha=alpha,
+    )
+    prop_to_hospital_2 = 1 - prop_to_hospital_1
+
+    proportion_within_target_1 = (
+        proportion_within_target_using_markov_state_probabilities(
+            lambda_2=lambda_2 * prop_to_hospital_1,
+            lambda_1=lambda_1_1,
+            mu=mu_1,
+            num_of_servers=num_of_servers_1,
+            threshold=threshold_1,
+            system_capacity=system_capacity_1,
+            buffer_capacity=buffer_capacity_1,
+            class_type=None,
+            target=target,
+        )
+    )
+    proportion_within_target_2 = (
+        proportion_within_target_using_markov_state_probabilities(
+            lambda_2=lambda_2 * prop_to_hospital_2,
+            lambda_1=lambda_1_2,
+            mu=mu_2,
+            num_of_servers=num_of_servers_2,
+            threshold=threshold_2,
+            system_capacity=system_capacity_2,
+            buffer_capacity=buffer_capacity_2,
+            class_type=None,
+            target=target,
+        )
+    )
+    if alternative_utility:
+        utility_1 = proportion_within_target_1
+        utility_2 = proportion_within_target_2
+    else:
+        utility_1 = -((proportion_within_target_1 - 0.95) ** 2)
+        utility_2 = -((proportion_within_target_2 - 0.95) ** 2)
+
+    return threshold_1, threshold_2, prop_to_hospital_1, utility_1, utility_2
+
+
+def compute_tasks(tasks, processes):
+    """
+    Compute all dask tasks
+    """
+    if processes is None:
+        out = da.compute(*tasks, scheduler="single-threaded")
+    else:
+        out = da.compute(*tasks, num_workers=processes)
+    return out
+
+
+def build_matrices_from_computed_tasks(computed_tasks, N_1, N_2):
+    """
+    Using the computed tasks builds the utility matrix of the row and the column
+    players and the routing matrix.
+
+    Parameters
+    ----------
+    computed_tasks : tuple
+        A tuple of tuples of the form (i, j, R[i,j], A[i,j], B[i,j])
+    N_1 : int
+        The number of rows for all matrices
+    N_2 : int
+        The number of columns for all matrices
+
+    Returns
+    -------
+    numpy array, numpy array, numpy array
+        The routing matrix and the two payoff matrices
+    """
+    routing_matrix = np.zeros((N_1, N_2))
+    utility_matrix_1 = np.zeros((N_1, N_2))
+    utility_matrix_2 = np.zeros((N_1, N_2))
+
+    for row, col, routing_entry, utility_1_entry, utility_2_entry in computed_tasks:
+        routing_matrix[row - 1, col - 1] = routing_entry
+        utility_matrix_1[row - 1, col - 1] = utility_1_entry
+        utility_matrix_2[row - 1, col - 1] = utility_2_entry
```

Comment:
```suggestion
    for threshold_1, threshold_2, routing_entry, utility_1_entry, utility_2_entry in computed_tasks:
        row_index, col_index =  threshold_1 - 1, threshold_2 - 1
        routing_matrix[row_index, col_index] = routing_entry
        utility_matrix_1[row_index, col_index] = utility_1_entry
        utility_matrix_2[row_index, col_index] = utility_2_entry
```

I am 60% sure my suggestion is an improvement but I am happy if you ignore it.

Reply:
That looks reasonable to me.